File: 1114.py
class Foo:
    def __init__(self):
        self.x = 0
        pass

    def first(self, printFirst: 'Callable[[], None]') -> None:
        printFirst()
        self.x += 1

    def second(self, printSecond: 'Callable[[], None]') -> None:
        while self.x < 1:
            pass
        printSecond()
        self.x += 1

    def third(self, printThird: 'Callable[[], None]') -> None:
        while self.x < 2:
            pass
        printThird()

File: 1114.java
class Foo {

    private volatile int count = 0;

    public Foo() {
    }

    public void first(Runnable printFirst) throws InterruptedException {
        printFirst.run();
        count++;
    }

    public void second(Runnable printSecond) throws InterruptedException {
        while (count < 1) {
        }
        printSecond.run();
        count++;
    }

    public void third(Runnable printThird) throws InterruptedException {
        while (count < 2) {
        }
        printThird.run();
    }
}